nft/payment_obligation_integration_test.go

```diff
@@ -66,21 +69,42 @@ func TestPaymentObligationService_mint(t *testing.T) {
 		},
 	})
 	assert.Nil(t, err, "should not error out when creating invoice model")
-	modelUpdated, err := invoiceService.Create(contextHeader, model)
+	modelUpdated, txID, err := invoiceService.Create(contextHeader, model)
+	confirm, err := waitTillSuccessOrError(txID, txService)
+	assert.Nil(t, err)
+	assert.True(t, confirm)
 
 	// get ID
 	ID, err := modelUpdated.ID()
 	assert.Nil(t, err, "should not error out when getting invoice ID")
 	// call mint
 	// assert no error
-	confirmations, err := payOb.MintNFT(
+	resp, err := payOb.MintNFT(
+		ccommon.DummyIdentity,
 		ID,
 		cfg.GetContractAddress("paymentObligation").String(),
 		"0xf72855759a39fb75fc7341139f5d7a3974d4da08",
 		[]string{"invoice.gross_amount", "invoice.currency", "invoice.due_date", "collaborators[0]"},
 	)
 	assert.Nil(t, err, "should not error out when minting an invoice")
-	tokenConfirm := <-confirmations
-	assert.Nil(t, tokenConfirm.Err, "should not error out when minting an invoice")
-	assert.NotNil(t, tokenConfirm.TokenID, "token id should be present")
+	assert.NotNil(t, resp.TokenID, "token id should be present")
+}
+
+func waitTillSuccessOrError(txID uuid.UUID, txService transactions.Service) (bool, error) {
+	status := transactions.Pending
+	for status == transactions.Pending {
```

Comment:
this should be a function in the `txService`
also please add a timer instead of a busy loop

Reply:
This is test and shouldn't be of an issue in prod IMO